code/modules/scrap/scrap_beacon.dm

```diff
@@ -0,0 +1,48 @@
+/obj/structure/scrap_beacon
+	name = "Scrap Beacon"
+	desc = "This machine generates directional gravity rays which catch trash orbiting around."
+	icon = 'icons/obj/structures/scrap/scrap_beacon.dmi'
+	icon_state = "beacon0"
+	anchored = TRUE
+	density = TRUE
+	layer = MOB_LAYER + 1
+	var/summon_cooldown = 1200
+	var/impact_speed = 3
+	var/impact_prob = 100
+	var/impact_range = 2
+	var/last_summon = -3000
+	var/active = 0
+
+/obj/structure/scrap_beacon/attack_hand(mob/user)
```

Comment:
No ide aif you guys are going to ever port it but if you ever want to port component signals, stuff like attack_hand will need to call ..() for them to work right